### PR TITLE
[DROOLS-1764] use (un)trusted xstream instances from XStreamUtils

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java [hunk 1]
```diff
@@ -222,7 +179,7 @@ public static String formatXml(String xml){
      }
     
     private XStream newXStream() {
-        XStream xStream = createXStream( staxDriver, Definitions.class.getClassLoader() );
+        XStream xStream = createNonTrustingXStream( staxDriver, Definitions.class.getClassLoader() );
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java:182]: I suppose this means for DMN we need to create some callbacks in order to allow for the end-user to provide whitelisting

**Review by tarilabs — APPROVED** [review]
...and I assume in general the whitelisting API will also be required for Drools in general ?

File: kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java (hunk 1 above)
Comment (mariofusco) [reply kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java:182]: Yes, this is what you told me you were already doing. Is this correct?

Comment (tarilabs) [reply kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java:182]: unfortunately not, I've said that is what we _should_ be doing :)

**Pull request closed without merging**